### PR TITLE
fix macOS compile issue for `timespec_get`

### PR DESCRIPTION
**Pull request by StarWitch**
As seen here (thanks @Akaricchi): https://stackoverflow.com/questions/32803754/what-is-the-darwin-c-level-c-preprocessor-symbol

Also adds some useful tips in the readme for building on Linux/macOS, and adds *.swp files as ignored (common for vim users).

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -220,6 +220,7 @@ if macos_app_bundle
     datadir = join_paths(bundle_dir, 'Contents', 'Resources')
     bindir = join_paths(bundle_dir, 'Contents', 'MacOS')
     config.set('TAISEI_BUILDCONF_RELATIVE_DATA_PATH', true)
+    config.set('TAISEI_DARWIN_SOURCE', true)
```
Comment (Akaricchi) [inline meson.build]: This should be called `TAISEI_BUILDCONF_DARWIN_SOURCE` for consistency if we're doing it this way. It's also under the wrong conditional, because `macos_app_bundle` can be `false` if the `macos_bundle` option is disabled (you get a traditional unix-like directory structure instead of an app bundle in that case).

This should be under an `if host_machine.system() == 'darwin'` clause — that'll cover macOS, iOS, and other Apple systems in case someone decides to make a port. Alternatively, I think it's fine to just check for the `__APPLE__` macro in `util/compat.h` instead.

File: src/util/compat.h [hunk 2]
```diff
@@ -15,6 +15,11 @@
 #define _POSIX_C_SOURCE 200809L
 #endif
 
+// Apple doing weird stuff again
```
Comment (Akaricchi) [inline src/util/compat.h]: This is probably worth briefly explaining, e.g. "_POSIX_C_SOURCE makes some standard definitions disappear from Apple's headers unless this is defined". Expletives permitted.

File: README.rst [hunk 3]
```diff
@@ -31,6 +31,15 @@ Optional:
 -  SPIRV-Cross >= 2019-03-22 (for OpenGL ES backends)
 -  libshaderc (for OpenGL ES backends)
 
+Platform-Specific Instructions
```
Comment (Akaricchi) [inline README.rst]: Could the readme changes please go in a separate PR? I am very much in favor of documenting the dependency subprojects, though.

**Comment by StarWitch** [comment]
Done!

**Comment by Akaricchi** [comment]
Thank you!
